src/rpcmisc.cpp

```diff
@@ -206,6 +206,37 @@ UniValue validateaddress(const UniValue& params, bool fHelp)
 }
 
 
+class DescribePaymentAddressVisitor : public boost::static_visitor<UniValue>
+{
+public:
+    UniValue operator()(const libzcash::InvalidEncoding &zaddr) const { return UniValue(UniValue::VOBJ); }
+
+    UniValue operator()(const libzcash::SproutPaymentAddress &zaddr) const {
+        UniValue obj(UniValue::VOBJ);
+        obj.push_back(Pair("type", "sprout"));
+        obj.push_back(Pair("payingkey", zaddr.a_pk.GetHex()));
+        obj.push_back(Pair("transmissionkey", zaddr.pk_enc.GetHex()));
+#ifdef ENABLE_WALLET
+        if (pwalletMain) {
+            obj.push_back(Pair("ismine", pwalletMain->HaveSpendingKey(zaddr)));
+        }
+#endif
+        return obj;
+    }
+
+    UniValue operator()(const libzcash::SaplingPaymentAddress &zaddr) const {
+        UniValue obj(UniValue::VOBJ);
+        obj.push_back(Pair("type", "sapling"));
+#ifdef ENABLE_WALLET
+        if (pwalletMain) {
+            // TODO uncomment when #3273 is merged
+            // obj.push_back(Pair("ismine", pwalletMain->HaveSaplingSpendingKey(zaddr)));
```

Comment:
`HaveSaplingSpendingKey` will need to take the full viewing key and not a zaddr as an arg